GoRogue/MapGeneration/Region.cs

```diff
@@ -72,62 +118,21 @@ public Region(Lines.Algorithm algorithm, IComponentCollection? components, param
         /// <param name="corners">The corners of the region</param>
         public Region(params Point[] corners) : this(corners, Lines.Algorithm.DDA) { }
 
-        private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components)
+        /// <summary>
+        /// Returns a new Region using the provided area
+        /// </summary>
+        /// <param name="area">This region's Area</param>
+        /// <param name="components">A component collection for this region</param>
+        public Region(PolygonArea area, IComponentCollection? components = null)
         {
-            Area = new PolygonArea(ref corners, algorithm);
+            Area = area;
             GoRogueComponents = components ?? new ComponentCollection();
             GoRogueComponents.ParentForAddedComponents = this;
         }
 
-        #region Properties
-
-        /// <summary>
-        /// The left-most X-value of the region's four corners
-        /// </summary>
-        public int Left => Area.Left;
-
-        /// <summary>
-        /// The right-most X-value of the region's four corners
-        /// </summary>
-        public int Right => Area.Right;
-
-        /// <summary>
-        /// The top-most Y-value of the region's four corners
-        /// </summary>
-        public int Top => Area.Top;
-
-        /// <summary>
-        /// The bottom-most Y-value of the region's four corners
-        /// </summary>
-        public int Bottom => Area.Bottom;
-
-        /// <summary>
-        /// How Wide this region is
-        /// </summary>
-        public int Width => Area.Width;
-
-        /// <summary>
-        /// how tall this region is
-        /// </summary>
-        public int Height => Area.Height;
-
-        /// <summary>
-        /// The Center point of this region
-        /// </summary>
-        public Point Center => Area.Center;
-
-        /// <inheritdoc />
-        public Rectangle Bounds => Area.Bounds;
-
-        /// <inheritdoc />
-        public int Count => Area.Count;
-
-        /// <inheritdoc />
-        public Point this[int index] => Area[index];
-        #endregion
-
-        //Functions to access information about regions
-        #region Data Access Functions
+        private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components)
+            : this(new PolygonArea(ref corners, algorithm))
```

Comment:
```suggestion
            : this(new PolygonArea(ref corners, algorithm), components)
```

Missing components forward.